servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancer.java

```diff
@@ -507,10 +513,38 @@ private void sequentialUpdateUsedHosts(List<PrioritizedHostImpl<ResolvedAddress,
             host.loadBalancingWeight(host.serviceDiscoveryWeight());
         }
         nextHosts = priorityStrategy.prioritize(nextHosts);
+        nextHosts = makeSubset(nextHosts);
         this.hostSelector = hostSelector.rebuildWithHosts(nextHosts);
         loadBalancerObserver.onHostSetChanged(Collections.unmodifiableList(nextHosts));
     }
 
+    private List<PrioritizedHostImpl<ResolvedAddress, C>> makeSubset(
+            final List<PrioritizedHostImpl<ResolvedAddress, C>> nextHosts) {
+        if (nextHosts.size() <= subsetSize) {
+            return nextHosts;
+        }
+
+        // We need to sort, and then return the list with the subsetSize number of healthy elements.
+        ArrayList<PrioritizedHostImpl<ResolvedAddress, C>> result = new ArrayList<>(nextHosts);
+        result.sort((a, b) -> Long.compare(a.randomSeed, b.randomSeed));
+
+        // We don't want to consider the unhealthy elements to be a part of our subset, so we're going to grow it
+        // to account for un-health endpoints. However, we need to know how many that is.
+        for (int i = 0, healthyCount = 0; i < result.size(); i++) {
+            if (result.get(i).isHealthy()) {
+                ++healthyCount;
+                if (healthyCount == subsetSize) {
+                    // Trim elements after i to form the subset.
+                    while (result.size() > i + 1) {
+                        result.remove(result.size() - 1);
```

Comment:
If we anticipate this path to execute for large lists, another option to consider is taking `result.subList` instead of a removal loop

Reply:
I'd like to leave it this way for now. This particular operation should be O(N) so it should be cheap enough considering we're doing a O(n log n) sort up above. In contrast, the added sublist indirection will have to do additional bounds checks for every `List.get(int)` call which is expected to happen much more often.